packages/atlas/src/views/viewer/VideoView/CommentThread.tsx

```diff
@@ -31,24 +65,75 @@ export const CommentThread: React.FC<CommentThreadProps> = ({
         setRepliesOpen={setRepliesOpen}
         isRepliesOpen={repliesOpen}
         setHighlightedCommentId={setHighlightedCommentId}
+        userReactions={userReactionsLookup && commentId ? userReactionsLookup[commentId] : undefined}
         {...commentProps}
         isReplyable={true}
+        onReplyPosted={setNewReplyId}
       />
-      {repliesOpen &&
-        !!repliesCount &&
-        (!replies
-          ? placeholderItems.map((_, idx) => <Comment key={idx} indented />)
-          : replies?.map((comment, idx) => (
+      {linkedReplyId && !repliesOpen && (
+        <Comment
+          key={`${commentId}-linked-reply`}
+          highlighted={linkedReplyId === highlightedCommentId}
+          commentId={linkedReplyId}
+          video={video}
+          indented
+          setHighlightedCommentId={setHighlightedCommentId}
+          userReactions={userReactionsLookup ? userReactionsLookup[linkedReplyId] : undefined}
+          isReplyable={false}
+        />
+      )}
+      {repliesOpen && (
+        <>
+          {replies?.map((comment, idx) => (
+            <Comment
+              key={`${comment.id}-${idx}`}
+              highlighted={comment.id === highlightedCommentId}
+              commentId={comment.id}
+              video={video}
+              indented
+              setHighlightedCommentId={setHighlightedCommentId}
+              userReactions={userReactionsLookup ? userReactionsLookup[comment.id] : undefined}
+              isReplyable={false}
+            />
+          ))}
+          {placeholderItems.map((_, idx) => (
+            <Comment key={idx} indented />
+          ))}
+          {repliesLeftToLoadCount > 0 ? (
+            <LoadMoreRepliesButton
+              variant="tertiary"
+              onClick={handleLoadMore}
+              icon={<SvgActionChevronB />}
+              iconPlacement="right"
+            >
+              Load more replies ({repliesLeftToLoadCount})
+            </LoadMoreRepliesButton>
+          ) : null}
+          {newReplyId && !allRepliesContainNewReply ? (
+            newReply ? (
               <Comment
-                key={`${comment.id}-${idx}`}
-                highlighted={comment.id === highlightedCommentId}
-                commentId={comment.id}
+                key={newReply?.id}
+                highlighted={newReply.id === highlightedCommentId}
+                commentId={newReply.id}
                 video={video}
                 indented
                 setHighlightedCommentId={setHighlightedCommentId}
+                userReactions={userReactionsLookup ? userReactionsLookup[newReply.id] : undefined}
                 isReplyable={false}
               />
-            )))}
+            ) : (
+              <Comment indented />
```

Comment:
Hmm, maybe it's just me, but this was a little confusing for me. I had to check why we have a blank Comment component without all the props. Can we add a `loading` prop to Comment, just for the sake of clarity?

Reply:
I'm not sure if `loading` prop makes sense for `Comment` but added a comment (😉) here to explain why we display this